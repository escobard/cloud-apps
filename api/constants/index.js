const dbkeys = require("./dbkeys"),
  routes = require("./routes")

module.exports = {
  dbkeys,
  routes
}
