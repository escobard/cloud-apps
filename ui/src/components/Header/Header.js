import React from "react";

import "./Header.scss";

/** Header Component
 * @name Header
 * @param {string} id, inherited ID from parent
 * @return {Component} Header
 * */

const Header = ({ id }) => (
  <nav id={`${id}-navigation`}>
    <i aria-hidden="true" className="bars big icon" />
    <h1>
      Your <br />
      Notes
    </h1>
    <p>Nov 5, 2019</p>
  </nav>
);

export default Header;
