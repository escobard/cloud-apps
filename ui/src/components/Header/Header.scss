@import "../../styles/variables";

/* start by selecting the parent ID */
#navigation{
  /* will need a higher res image for screens larger than 1800px*/
  background: url("../../assets/images/nav-background.jpg");
  background-size: cover;
  background-position: 100% -50vh;
  background-attachment: fixed;
  background-repeat: no-repeat;
  height: 270px;
  padding: $space-xlrg;
  h1,p, i{
    color: $white;
  }
  h1{
    font-size: $font-size-mega;
    margin-bottom: $space-xlrg;
  }
  h1, p{
    font-family: $font-family-default;
    font-weight: 300;
  }
}
