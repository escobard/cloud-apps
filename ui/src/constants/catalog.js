import { todayDate } from "utils/todayDate";

export const footer = {
  noNotes: "NO NOTES",
  withNotes: "COMPLETED"
};

export const header = {
  title: "Your Notes",
  todayDate: todayDate()
};

export const notes = {
  noNotes: {
    subject: "No Notes",
    note: "Add a note by clicking on the + icon below.",
    icon: {
      image: "exclamation",
      label: "No Notes"
    }
  },
  apiError: {
    subject: "API error",
    icon: {
      image: "exclamation",
      label: "API error"
    }
  }
};
