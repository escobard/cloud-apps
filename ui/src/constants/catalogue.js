import { todayDate } from "utils/todayDate";

export const footer = {
  noNotes: "NO NOTES",
  withNotes: "COMPLETED"
};

export const header = {
  title: "Your Notes",
  todayDate: todayDate()
}