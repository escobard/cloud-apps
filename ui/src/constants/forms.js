export const addNoteFields = [
  {
    name: "subject",
    label: "Subject",
    placeholder: "Enter a subject for your note",
    value: "",
    errors: {
      // expand this later with future error strings
    }
  },
  {
    name: "note",
    label: "Note *",
    placeholder: "Enter a description for your note (required)",
    value: "",
    errors: {}
  }
];