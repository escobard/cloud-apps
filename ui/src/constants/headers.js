export const headers = { "Access-Control-Allow-Origin": "*" };
