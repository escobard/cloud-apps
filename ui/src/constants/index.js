import { addNoteFields } from "./forms";
import { apiRoutes } from "./urls";
import { headers } from "./headers";
import { notes, footer, header } from "./catalog";

export { apiRoutes, addNoteFields, headers, notes, footer, header };
