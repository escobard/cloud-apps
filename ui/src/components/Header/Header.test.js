import { header } from "constants/catalog";
import Header from "./index";

const props = {
  id: "test",
  date: "February 23, 2020"
};

const { title, todayDate } = header;

describe("Header", () => {
  afterAll(() => {
    cleanup();
  });

  it(">> snapshot is up to date", () => {
    const { container } = render(<Header {...props} />);
    expect(container).toMatchSnapshot();
  });
  it(">> title and date are rendered", () => {
    props.date = todayDate;
    const { getByText } = render(<Header {...props} />);
    expect(getByText(title));
    expect(getByText(todayDate));
  });
});
