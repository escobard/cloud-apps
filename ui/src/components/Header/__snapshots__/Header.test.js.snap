// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header >> snapshot is up to date 1`] = `
<div>
  <nav
    id="test-navigation"
  >
    <i
      aria-hidden="true"
      class="bars big icon"
    />
    <h1>
      Your Notes
    </h1>
    <p />
  </nav>
</div>
`;
